src/Stratis.Bitcoin.Features.Consensus/StakeItem.cs

```diff
@@ -8,6 +8,8 @@ public class StakeItem
 
         public BlockStake BlockStake;
 
+        public ProvenBlockHeader ProvenBlockHeader;
```

Comment:
Why adding it to stake item?

Reply:
`StakeItem` has all the necessary properties to help persist a `ProvenBlockHeader` item.

It also allows the `ProvenBlockHeaderRepository` to sort on `StakeItem.Height`, before saving to disk.  Also to check whether a `ProvenBlockHeader` is in `InStore` (on disk).  

Why do you ask - do you think there is a better approach then using this class ?

Reply:
The problem is you are mixing functionalities, ITs better you create a dedicated class for PH if not it will be confusing for the stake store code